Some i18n support

mcarcaso opened the pull request:
Adds the Message axiom and an android string generator that makes use of it.

mcarcaso committed:
Re-introduce the i18n stuff that I had in a different branch at one point.

mcarcaso committed:
Use foam code style on an if statement.

mcarcaso committed:
Remove awhile because it doesn't belong in this branch and doesn't have anything using it yet.

mcarcaso committed:
Change the android GenStrings tool to refine the axioms to turn themselves into string resources and fix typo in the demo script.

mcarcaso committed:
Move the GenStrings model to the top of the file.

mcarcaso committed:
Remove unused classes model property.

mcarcaso committed:
Merge branch 'master' into i18n

adamvy commented on src/foam/android/GenResources.js:
```diff
@@ -0,0 +1,71 @@
+ */
+
+foam.CLASS({
+  package: 'foam.android',
```
Might make more sense for things like this to live in foam.android.tools

I sure in the future we'll have foam.android.dao, foam.android.ui etc etc.

adamvy-google commented on src/foam/i18n/Messages.js:
```diff
@@ -0,0 +1,83 @@
+      class: 'AxiomArray',
+      of: 'foam.i18n.MessageAxiom',
+      adaptArrayElement: function(o, prop) {
+        return foam.lookup(prop.of).create(o);
```
I believe this is the default adapt of AxiomArray, so you can leave it out.

mcarcaso committed:
Move the android string files to the foam.android.tools package and remove a useless adapt on MessageAxiom.

mcarcaso committed:
Merge branch 'master' into i18n

mcarcaso committed:
Merge branch 'RolledBackModelDAOs' into i18n

mcarcaso committed:
Merge branch 'RolledBackModelDAOs' into i18n

mcarcaso commented:
Sorry I forgot about this pull request. I'm pretty sure I addressed Adam's comments way back when. I rebased this on the RolledBackModelDAOs branch because it depends on the behaviour of the rolled back DAOs to get the right dependencies in place.

mcarcaso committed:
Merge branch 'master' into i18n

jacksonic commented:
Was GoogleBot confused by the RolledBackModelDAOs base branch swap? I don't see any non-mcarcaso commits now.

kgrgreer commented:
I think it was confused because I hadn't set my GIT username property on my
new machine.

On 20 April 2017 at 08:10, Jackson Stone <notifications@github.com> wrote:

> Was GoogleBot confused by the RolledBackModelDAOs base branch swap? I
> don't see any non-mcarcaso commits now.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/foam-framework/foam2/pull/237#issuecomment-295710505>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHHXs4EjsOFxlibW6cF1KUJlchA2JMklks5rx0smgaJpZM4L6F7i>
> .
>
